Luna-Interactive/catastrophe
# Special Game Mode!

## Description of change
This PR adds a small little special mode! When a team picks up the flag, players in other teams become teammates (so can't shoot each other). As soon as the flag is dropped, players go back to their old teams.

The game mode is already turned on (See `GameInfo.specialGameMode`). If you want to turn it off, you have to set the boolean to false.

@corentinartaud will take care of letting the server choose the game mode in the menu (either in this PR or another).

## How to reproduce the changes
1. Make sure the special game mode is on
2. Run three instances of the game. Get into a game
3. Pick up the flag with a player. You should see the other players' color rings change to yellow. The yellow players cannot damage each other.
4. Either drop the flag or kill the flag-carrier. you should see the team colors go back to normal.

## Screenshots or GIFs
My laptop is slow af, I can barely run three instances of the game. No screenshots. 😛
